docs/content/policy-cheatsheet.md

```diff
@@ -268,6 +268,98 @@ z := {"a": true, "b": "foo", "c": 4}
 object.union(y, x) == z
 ```
 
+#### Parse array of sets into nested objects
+
+This is an example of parsing an array of sets that have a known structure into nested objects.
+
+```live:rules/parse_array_to_nested_objs:query:read_only
+entity_array = [
+  [
+    "productName1",
+    "entityType1",
+    "entityId1",
+    "duck",
+    {
+      "name": "Yolanda"
+    }
+  ],
+  [
+    "productName1",
+    "entityType1",
+    "entityId2",
+    "weave",
+    {
+      "name": "Zoe"
+    }
+  ],
+  [
+    "productName1",
+    "entityType2",
+    "entityId3",
+    "wink",
+    {
+      "name": "Audrey"
+    }
+  ],
+  [
+    "productName2",
+    "entityType1",
+    "entityId4",
+    "wave",
+    {
+      "name": "Beryl"
+    }
+  ],
+  [
+    "productName2",
+    "entityType2",
+    "entityId5",
+    "wander",
+    {
+      "name": "Cynthia"
+    }
+  ],
+  [
+    "productName2",
+    "entityType2",
+    "entityId6",
+    "waft",
+    {
+      "name": "Daniela"
+    }
+  ]
+]
+
+nestedobjs = objs {
+  objs := parse_to_nestedobjs(entity_array)
+}
+```
+
+```live:rules/parse_array_to_nested_objs:module:read_only
+parse_to_nestedobjs(arr) = res {
+  res := { product_name : entity_types |
+    product_name := arr[_][0]
+    entity_types := { entity_type : entities |
+      arr[idx1][0] == product_name
+      entity_type := arr[idx1][1]
+      entities := { entity_id : entity |
+        arr[idx2][0] == product_name
+        arr[idx2][1] == entity_type
+        entity_id := arr[idx2][2]
+        entity := {
+          "action": arr[idx2][3],
+          "metadata": arr[idx2][4]
+        }
+      }
+    }
+  }
+}
```

Comment:
It might be good to add some inline comments just walking through the like logic of what this is doing so people can not only copy and re-use but (hopefully) understand and iterate/adapt it for their use-case.

Reply:
I've condensed the query and added comments to the function, do they look OK?

Reply:
I like comments, only nit would be do move them to the side so its like policy on the left and comments on the right like:

```
do_the_thing() {		# Helper to do the thing
      a + b = 3			# That makes sure it equals 3 for some reasons 
      another_thing		# Finished up with another thing because X
}
```

Reply:
Good idea. It looked a bit busy to me, but adding blank lines didn't really help.